code/tailer/datastripe.q

```diff
@@ -118,33 +112,40 @@ upserttopartition:{[dir;tablename;keycol;enumdata;nextp]
 savetablesoverperiod:{[dir;tablename;nextp;lasttime]
     /- function to get keycol for table from access table
     keycol:`sym^.wdb.tablekeycols tablename;
-
     /- get distinct values to partition table on
     partitionlist: ?[tablename;();();(distinct;keycol)];
-
     /- enumerate and then split by keycol
     symdir:` sv dir,.proc.procname;
     enumkeycol: .Q.en[symdir;?[tablename;enlist (<;`time;nextp);0b;()]];
     splitkeycol: {[enumkeycol;keycol;s] ?[enumkeycol;enlist (=;keycol;enlist s);0b;()]}[enumkeycol;keycol;] each partitionlist;
-
     /-upsert table to partition
-    upserttopartition[dir;tablename;keycol;;nextp] each splitkeycol where 0<count each splitkeycol; 
-
+    upserttopartition[dir;tablename;keycol;;nextp] each splitkeycol where 0<count each splitkeycol;
     /- delete data from last period
-    .[.ds.deletetablebefore;(tablename;`time;lasttime)];
-    
+    ![;();0b;`symbol$()]each tablename; 
     /- run a garbage collection (if enabled)
     .gc.run[];
     };
 
-savealltablesoverperiod:{[dir;nextp;lasttime]
+savealltablesoverperiod:{[]
     /- function takes the tailer hdb directory handle and a timestamp
     /- saves each table up to given period to their respective partitions
-    savetablesoverperiod[dir;;nextp;lasttime]each .wdb.tablelist[];
+    /- totals calculates the row count of tables in .wdb.tablelist[] 
+    totals:{count get x}each .wdb.tablelist[];
+    /- If a totals value is greater than rowthresh which is the rowcount for saving down tables
+    if[max totals>.wdb.rowthresh;
+        .lg.o[`save;"Saving ",(", " sv string .wdb.tablelist[] where totals>.wdb.rowthresh)," table(s)"];
+        savetablesoverperiod[.ds.td;;.z.p;(.z.p-`second$.wdb.period)]each (.wdb.tablelist[] where totals>.wdb.rowthresh);
     /- trigger reload of access tables and intradayDBs in all tail reader processes
-    .tailer.dotailreload[`]};
+    .tailer.dotailreload[`]
+    ];
+    /- if totals are below rowthresh log message indicates no tables saved
+    if[min totals<.wdb.rowthresh;
```

Comment:
as above but

```suggestion
    if[not any totals<.wdb.rowthresh;
```

or

```suggestion
    if[all totals>.wdb.rowthresh;
```

I would also move this conditional to the start of the function & return early from it (i.e. `:();`), then have the code that does the save *outside* any conditional (because obviously if the first conditional is false, the other is true, so don't necessarily need to specify it)

Reply:
This has been restructured to only use one conditional, the semantics of which has been improved as per this suggestion to use `all`